uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/App.java

```diff
@@ -116,6 +146,19 @@ public void launch() {
         CommonUtils.launchApp(packageName);
     }
 
+    @Override
+    public void update() {
+        String appHint;
+        if (updateAppInstaller instanceof LocalApkInstaller && !TextUtils.isEmpty(localApkFileName)) {
```

Comment:
What if the initial install is also done using the same `localApkFileName`?

Reply:
This is not possible in update scenarios. I have pushed changes in this (https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1640/files) PR to make sure that in case of update scenarios - We extend AbstractMsalBrokerUpdateTest in all update scenarios to first install old apk and then install the latest one when update function is called